plugins/inputs/elasticsearch/elasticsearch.go

```diff
@@ -267,6 +286,22 @@ func (e *Elasticsearch) nodeStatsUrl(baseUrl string) string {
 	return fmt.Sprintf("%s/%s", url, strings.Join(e.NodeStats, ","))
 }
 
+func (e *Elasticsearch) gatherNodeID(info *serverInfo, url string) error {
```

Comment:
Can you have this and also `setCatMaster` return just the nodeID and the masterID, we can then put them together in the calling function and avoid using out parameters.

When updating the struct field, we will need a mutex to avoid a race error since these are being called in a goroutine.